plugins/gatsby-theme-iterative-docs/src/components/Documentation/Markdown/index.tsx

```diff
@@ -50,11 +50,21 @@ const Details: React.FC<{ slugger: GithubSlugger }> = ({
     firstChild.props.children.length - 1
   ) as ReactNode[]
 
-  let slug = slugger.slug(triggerChildren.toString())
-  if (slug[0] === '️') {
-    slug = slug.slice(1)
+  const title = (triggerChildren as any[]).reduce((acc, cur) => {
+    return (acc +=
+      typeof cur === 'string'
+        ? cur
+        : typeof cur === 'object'
+        ? cur?.props?.children?.toString()
+        : '')
+  }, '')
+
+  let slug = slugger.slug(title)
+  if (slug.includes('️')) {
+    slug = slug.replaceAll('️', '')
   }
-  const id = slug.startsWith('-') ? slug.slice(1) : slug
+  slug = slug.startsWith('-') ? slug.slice(1) : slug
+  const id = slug.endsWith('-') ? slug.slice(0, -1) : slug
```

Comment:
Maybe a regex-based replacement would be good in order to catch multiple dashes on the ends of the string, but admittedly it's a little unlikely that titles with the current format would do so.